### PR TITLE
Fix validate_btrfs

### PR DESCRIPTION
**Pull request by grisu48**
Fix for wrongly assuming 20GB disk size in the validate_btrfs
container tests.

- Related ticket: https://progress.opensuse.org/issues/93339
- Verification run: http://duck-norris.qam.suse.de/tests/6570

### CODE REVIEW

File: tests/containers/validate_btrfs.pm [hunk 1]
```diff
@@ -55,12 +63,16 @@ sub _test_btrfs_device_mgmt {
...
+    script_run("df");
+    my $var_free = script_output('df 2>/dev/null | grep /var | awk \'{print $4;}\'');
+    # /var is using its own partition. Create file in the container enough to fill up the partition up to 99%
+    my $fill = int($var_free * 0.99 * 1024);    # df returns the size in KiB
+    $rt->up('huge_image', keep_container => 1, cmd => "fallocate -l $fill bigfile.txt");
     validate_script_output "df -h --sync|grep var", sub { m/\/dev\/vda.+\s+(9[7-9]|100)%/ };
     # partition should be full
-    validate_script_output "btrfs fi df $dev_path", sub { m/^Data.+total=8.*GiB, used=8.*GiB/ };
+    my ($total, $used) = _btrfs_fi("/var");
+    die "partitiion should be full" unless (int($total) == int($used));
```
Comment (jlausuch) [inline tests/containers/validate_btrfs.pm:75]:
```suggestion
    die "partition should be full" unless (int($total) == int($used));
```

File: tests/containers/validate_btrfs.pm [hunk 2]
```diff
@@ -55,12 +63,16 @@ sub _test_btrfs_device_mgmt {
     my $btrfs_head = '/tmp/subvolumes_saved';
     record_info "test btrfs";
     script_run("df -h");
-    # /var is using its own partition which is size of 10G. Create file in the container
-    # enough to fill up the partition up to ~99%
-    $rt->up('huge_image', keep_container => 1, cmd => 'fallocate -l 9149000KiB bigfile.txt');
+    # Determine the remaining size of /var
+    script_run("df");
```
Comment (jlausuch) [inline tests/containers/validate_btrfs.pm:67]: is this for debug purposes only? Maybe we could dump the output with `record_info` or something

**Comment by jlausuch** [comment]
Looks like a very good and generic solution. LGTM besides minor comments. 
Invalid VR?

File: tests/containers/validate_btrfs.pm (hunk 2 above)
Comment (grisu48) [reply tests/containers/validate_btrfs.pm]: Ah damn, this was forgotten :sweat_smile: Cleaned it up, thanks for the note!

**Comment by grisu48** [comment]
PR and VR are updated.

**Review by jlausuch — APPROVED** [review]

**Pull request merged**